app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,87 @@
+class RecipesController < ApplicationController
+  before_action :set_user_recipes, only: %i[show update destroy]
+  before_action :set_user_recipe, only: %i[show update]
+
+  def index
+    @recipes = current_user.recipes
+  end
+
+  def public_recipes
+    @recipes = Recipe.where("public = 'true' ")
+  end
+
+  def general_shopping_list
+    @foods = current_user.foods
+    @total_price = 0
+    @foods.each do |food|
+      @total_price += food.price * food.food_quantity
```

Comment:
- [ ] The controllers should be concerned with HTTP verbs. (thin controller rule) By that, I mean that a controller’s responsibility should be focused on web requests. Other calculations should be handled on the model.